modules/core/src/main/scala/sangria/schema/AstSchemaMaterializer.scala

```diff
@@ -296,12 +301,45 @@ class AstSchemaMaterializer[Ctx] private (
       scalarTypeExtensionDefs.flatMap(
         validateExtensionsAst[ScalarType[_], ast.ScalarTypeDefinition](_, "scalar")),
       unionTypeExtensionDefs.flatMap(
-        validateExtensionsAst[UnionType[_], ast.UnionTypeDefinition](_, "union"))
+        validateExtensionsAst[UnionType[_], ast.UnionTypeDefinition](_, "union")),
+      interfaceTypeDefs.flatMap(validateInterfaceType)
     )
 
     nestedErrors.flatten
   }
 
+  private def validateInterfaceType(
+      interfaceType: ast.InterfaceTypeDefinition): Vector[Violation] = {
+
+    def go(
+        validating: ast.InterfaceTypeDefinition,
+        visited: Set[ast.InterfaceTypeDefinition]): Vector[Violation] = {
+
+      val allInterfaceDefs: Vector[ast.InterfaceTypeDefinition] = (interfaceTypeExtensionDefs
+        .find(ie => ie.name == validating.name)
+        .map(ie => ie.interfaces)
+        .getOrElse(Vector.empty) ++ validating.interfaces).flatMap { i =>
```

Comment:
nitpick: using `.fold(Vector.empty[NamedType])(ie => ie.interfaces) ` instead of map and getOrElse

Reply:
Fixed